make/tests

```diff
@@ -1,67 +1,57 @@
 N_TESTS ?= 100 # override this in make/local. If <= 0, N_TESTS + 1 is interpreted as the number of batches to group the tests into
 
 ##
-# LIBGTEST is the google test library
 # GTEST_MAIN is the file that contains the google test
 ##
-LIBGTEST = test/libgtest.a
 GTEST_MAIN = $(GTEST)/src/gtest_main.cc
-CFLAGS_GTEST += -isystem $(GTEST)/include -isystem $(GTEST)
+GTEST_CXXFLAGS += -isystem $(GTEST)/include -isystem $(GTEST) -O$O
 
 ##
 # Build the google test library.
 ##
-$(LIBGTEST): $(LIBGTEST)(test/gtest.o)
+$(GTEST)/src/gtest-all.o: CXXFLAGS += $(GTEST_CXXFLAGS)
 
-test/gtest.o: $(GTEST)/src/gtest-all.cc
-	@mkdir -p test
-	$(COMPILE.cc) -O$O $(CFLAGS_GTEST) $< $(OUTPUT_OPTION)
+test/%$(EXE) : CXXFLAGS += $(GTEST_CXXFLAGS)
+test/%$(EXE) : test/%.o $(GTEST_MAIN) $(GTEST)/src/gtest-all.o
+	$(LINK.cpp) $^ $(LOADLIBES) $(LDLIBS) -o $@
 
-##
-# Rule for building a test
-##
-test/%_test.o : test/%_test.cpp
-	@mkdir -p $(dir $@)
-	$(COMPILE.cc) -O$O $(CFLAGS_GTEST) $< $(OUTPUT_OPTION)
 
-##
-# Rule for building a test executable
-##
-test/%$(EXE) : test/%.o $(LIBGTEST) $(LIBCVODES)
-	@mkdir -p $(dir $@)
-	$(LINK.cc) -O$O $(GTEST_MAIN) $< $(CFLAGS_GTEST) $(OUTPUT_OPTION) $(LIBGTEST) $(LDLIBS) $(LIBCVODES)
 
 
 ##
 # Rule for generating dependencies.
 ##
-.PRECIOUS: test/%_test.d
-test/%_test.d : test/%_test.cpp
+.PRECIOUS: test/%.d
+test/%.d : test/%.cpp
 	@mkdir -p $(dir $@)
 	@set -e; \
 	rm -f $@; \
-	$(COMPILE.cc) -O$O $(CFLAGS_GTEST) $(TARGET_ARCH) -MM $< > $@.$$$$; \
-	sed -e 's,\($(notdir $*)\)\(_test\)\.o[ :]*,$(dir $@)\1_test\$(EXE) $@ : ,g' < $@.$$$$ > $@; \
+	$(COMPILE.cpp) -MM $< > $@.$$$$; \
+	sed -e 's,\($(*F)\)\.o[ :]*,$(@D)/\1.o $@ : ,g' < $@.$$$$ > $@; \
 	rm -f $@.$$$$
 
-
-##
-# Adding a test for multiple translation units. If this fails,
+############################################################
+#
+# Test for multiple translation units. If this fails,
 # a new function is probably missing an inline
 ##
 
-test/unit/multiple_translation_units%.o : test/unit/multiple_translation_units%.cpp
-	$(COMPILE.cc) -O$O $< $(OUTPUT_OPTION)
-
-test/unit/libmultiple.so : test/unit/multiple_translation_units1.o test/unit/multiple_translation_units2.o
-	$(COMPILE.cc) -shared -fPIC $(OUTPUT_OPTION) $^
+test/unit/libmultiple.so : test/unit/libmultiple.so(test/unit/multiple_translation_units1.o test/unit/multiple_translation_units2.o)
 
+test/unit/multiple_translation_units_test.cpp : CXXFLAGS += -fPIC
+test/unit/multiple_translation_units_test.cpp : LDFLAGS += -shared
```

Comment:
Also, should this target be the executable, `test/unit/multiple_translation_units_test` (or the .so file) instead of the `.cpp` file (which is never generated)?